mlflow/gateway/utils.py

```diff
@@ -38,3 +56,97 @@ def kill_child_processes(parent_pid):
     _, still_alive = psutil.wait_procs(parent.children(), timeout=3)
     for p in still_alive:
         p.kill()
+
+
+def _is_valid_uri(uri: str):
+    """
+    Evaluates the basic structure of a provided gateway uri to determine if the scheme and
+    netloc are provided
+    """
+    try:
+        parsed = urlparse(uri)
+        return all([parsed.scheme, parsed.netloc])
+    except ValueError:
+        return False
+
+
+def _is_gateway_server_available(gateway_uri: str):
+    server_health_url = urljoin(gateway_uri, MLFLOW_GATEWAY_HEALTH_ENDPOINT)
+
+    try:
+        response = _get_gateway_response_with_retries("GET", server_health_url)
+        augmented_raise_for_status(response)
+    except HTTPError as http_err:
+        _logger.warning(f"There is not a gateway server running at {gateway_uri}. {http_err}")
+        return False
+    except Exception as err:
+        _logger.warning(
+            f"Unable to verify if a gateway server is healthy at {gateway_uri}. Error: {err}"
+        )
+        return False
+    else:
+        return True
+
+
+def _merge_uri_paths(paths: List[str]) -> str:
+    sanitized = [part.strip("/") for part in paths]
+    return "/".join(sanitized)
+
+
+def _resolve_gateway_uri(gateway_uri: Optional[str] = None) -> str:
+    gateway = gateway_uri or get_gateway_uri()
+    return gateway
+
+
+def set_gateway_uri(gateway_uri: str):
+    if not _is_valid_uri(gateway_uri):
+        raise MlflowException.invalid_parameter_value(
+            "The gateway uri provided is missing required elements. Ensure that the schema "
+            "and netloc are provided."
+        )
+
+    if not _is_gateway_server_available(gateway_uri):
+        raise MlflowException.invalid_parameter_value(
+            f"The gateway server cannot be verified at {gateway_uri}. Please verify that the "
+            "server has been started and that you are able to ping it."
+        )
```

Comment:
Is the server supposed to be running when setting the gateway URI?

Reply:
On Databricks, there won't be a health check endpoint, so we definitely can't block setting the gateway URI on this. I also agree with Haru's implication that we shouldn't require the gateway to be active when setting the URI

Reply:
I feel as though we should, right? I can't really see a user journey where someone is going to:
1. start a client
2. spend a few hours configuring the gateway server / start it somewhere
3. continue executing their interactive environment

It seems a forcing function to have a user ensure that the gateway server that they're setting a client to actually has to be running, but I feel like that makes a less confusing experience and helps to alert the user right as they create the client that there's an issue with the host configuration that they're setting up.

This can be easily changed though.